app/services/persister_task_service.rb

```diff
@@ -0,0 +1,54 @@
+class PersisterTaskService
+  def initialize(options)
+    @options = JSON.parse(options).deep_symbolize_keys
+    @upload_task = Task.find(@options[:category])
+  end
+
+  def process
+    @task = FullRefreshPersisterTask.create!(opts)
+    create_kafka_event
+
+    self
+  end
+
+  def create_kafka_event
+    CatalogInventory::Api::Messaging.client.publish_topic(
+      :service => "platform.catalog.persister",
+      :event   => "persister",
+      :payload => payload,
+      :headers => Insights::API::Common::Request.current_forwardable
+    )
+
+    Rails.logger.info("payload: #{payload}")
+    Rails.logger.info("event(Task.update) published to kafka.")
+  end
+
+  def opts
+    {}.tap do |options|
+      options[:tenant_id] = @upload_task.tenant_id
+      options[:source_id] = @upload_task.source_id
+      options[:input] = input
+      options[:state] = 'pending'
+      options[:status] = 'ok'
+    end
+  end
+
+  def input
+    {}.tap do |options|
+      options[:refresh_request_at] = @upload_task.created_at
+      options[:data_url] = @options[:url]
+      options[:size] = @options[:size]
+    end
+  end
+
+  def payload
+    input.merge!(:tenant_id => @upload_task.tenant_id, :source_id => @upload_task.source_id, :task_url => task_url)
+  end
+
+  def task_url
+    host = ENV.fetch("CATALOG_INVENTORY_INTERNAL_URL")
+    app_version = "v3.0"
```

Comment:
@hsong-rh We should come up with some constant for this so if we move to v3.1 or v4.0 we dont have to change this code. Maybe something like CURRENT_API_VERSION an environment variable that we can add when the app starts up 
Maybe set the env var here
https://github.com/RedHatInsights/catalog_inventory-api/blob/00bcaefd7e22366080a55dc52ed69cca34eb4829/config/application.rb#L45